src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,42 +1,86 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { useAppSelector } from '../../hooks';
+import { RootState } from '../../app/store';
+import { getUser } from '../../api';
+import { User } from '../../types/User';
 
-export const TodoModal: React.FC = () => {
-  return (
-    <div className="modal is-active" data-cy="modal">
-      <div className="modal-background" />
+interface TodoModalProps {
+  onClose: (isOpen: boolean) => void;
+}
 
-      <Loader />
+export const TodoModal: React.FC<TodoModalProps> = ({ onClose }) => {
+  const [user, setUser] = useState<User | null>(null);
+  const [isLoading, setIsLoading] = useState(false);
+  const todo = useAppSelector((state: RootState) => state.currentTodo);
 
-      <div className="modal-card">
-        <header className="modal-card-head">
-          <div
-            className="modal-card-title has-text-weight-medium"
-            data-cy="modal-header"
-          >
-            Todo #3
-          </div>
+  if (!todo) {
+    return null;
+  }
+
+  const { title, id, completed, userId } = todo;
+
+  // eslint-disable-next-line react-hooks/rules-of-hooks
+  useEffect(() => {
+    const fetchUser = async () => {
+      setIsLoading(true);
+      try {
+        const fetchedUser = await getUser(userId);
 
-          {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-          <button type="button" className="delete" data-cy="modal-close" />
-        </header>
+        setTimeout(() => {
```

Comment:
This is a performance/UX issue. It's not a good idea to use `setTimeout` in this way. This will introduce an artificial delay of 300ms before the user is shown, even if the data has already been loaded. It's better to remove the `setTimeout` and set the user data and loading state immediately when the data is loaded.